### PR TITLE
test: build new Ramps quotes 

### PR DESCRIPTION
**Pull request by bkirb**
## **Description**

This tests that the user can change options in the Build Quote screen and build new quotes. The new quotes should show the updated currency/token/fiat/etc. 

## **Related issues**

https://consensyssoftware.atlassian.net/browse/RAMPS-1933

## **Manual testing steps**

All Ramps tests are passing:
```
 PASS  e2e/specs/ramps/onramp.spec.js (185.993 s)
  smokeAssets Buy Crypto
    ✓ should select Region and Payment Method to see the Build Quote screen (17561 ms)
    ✓ should select a new region and check the quotes (44459 ms)
    ✓ should select a new token and check the quotes (35913 ms)
    ✓ should select a new currency and check the quotes (33345 ms)
    ✓ should select a new payment method and check the quotes (33233 ms)

Test Suites: 1 passed, 1 total
Tests:       5 passed, 5 total
```

## **Screenshots/Recordings**

Uploading Untitled.mov…

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by wachunei** [comment]
I'm concerned about the number of test ids being added, given that we know the default values, would not it be better to assert the presence of the default values by text [as described here](https://www.notion.so/metamask-consensys/TestIDs-c0ab66c3f44943cd936e0bdcf4d93c75)?

**Comment by bkirb** [comment]
> I'm concerned about the number of test ids being added, given that we know the default values, would not it be better to assert the presence of the default values by text [as described here](https://www.notion.so/metamask-consensys/TestIDs-c0ab66c3f44943cd936e0bdcf4d93c75)?

That makes sense. I removed most of the testIDs. Working on removing one last one for `amount-input`

**Comment by bkirb** [comment]
@chrisleewilcox I made some updates to address your feedback and tests are still passing ✅
https://github.com/MetaMask/metamask-mobile/pull/11675/commits/79d4ca10a5359e96dbbbf33e19ac9541ae138377

```
 PASS  e2e/specs/ramps/onramp.spec.js (227.889 s)
  smokeAssets Buy Crypto
    ✓ should select Region and Payment Method to see the Build Quote screen (16811 ms)
    ✓ should select a new currency and check the quotes (34342 ms)
    ✓ should select a new region and check the quotes (38020 ms)
    ✓ should select a new token and check the quotes (33718 ms)
    ✓ should select a new payment method and check the quotes (29683 ms)

Test Suites: 1 passed, 1 total
Tests:       5 passed, 5 total
```

File: e2e/pages/Ramps/QuoteView.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import TestHelpers from '../../helpers';
+import { QuoteSelectors } from '../../selectors/Ramps/Quote.selectors';
+
+class QuoteView {
+  get selectAQuoteLabel() {
+    return Matchers.getElementByText(QuoteSelectors.SELECT_A_QUOTE);
+  }
+
+  async dismiss() {
+    await TestHelpers.swipeByText('Select a Quote', 'down', 'fast', 0.5);
```
[the remark this answers is not in the document]
Comment (bkirb) [reply e2e/pages/Ramps/QuoteView.js]: Changed to `Gestures.swipe(...)`

File: e2e/selectors/Ramps/BuildQuote.selectors.js [hunk 2]
```diff
@@ -3,4 +3,6 @@ import enContent from '../../../locales/languages/en.json';
 export const BuildQuoteSelectors = {
   AMOUNT_TO_BUY_LABEL: enContent.fiat_on_ramp_aggregator.amount_to_buy,
   GET_QUOTES_BUTTON: enContent.fiat_on_ramp_aggregator.get_quotes,
+  CANCEL_BUTTON: enContent.navigation.cancel,
+  DONE_BUTTON: enContent.fiat_on_ramp_aggregator.done,
```
[the remark this answers is not in the document]
Comment (bkirb) [reply e2e/selectors/Ramps/BuildQuote.selectors.js]: Added `_TEXT` to both

File: e2e/selectors/Ramps/Quote.selectors.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import enContent from '../../../locales/languages/en.json';
+
+export const QuoteSelectors = {
+  SELECT_A_QUOTE: enContent.fiat_on_ramp_aggregator.select_a_quote,
```
[the remark this answers is not in the document]
Comment (bkirb) [reply e2e/selectors/Ramps/Quote.selectors.js]: Added `_TEXT`

**Pull request closed without merging**